cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -194,47 +218,69 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 		args.ContainerID, args.IfName, r)
 
 	// We will let the values in result struct guide us in terms of IP Address Family configured.
-	var v4Addr, v6Addr, addr *net.IPNet
 
-	// We don't support dual stack mode currently so it has to be either v4 or v6 mode.
-	if r.IPv4Addr != "" {
-		v4Addr = &net.IPNet{
-			IP:   net.ParseIP(r.IPv4Addr),
-			Mask: net.CIDRMask(32, 32),
-		}
-		addr = v4Addr
-	} else if r.IPv6Addr != "" {
-		v6Addr = &net.IPNet{
-			IP:   net.ParseIP(r.IPv6Addr),
-			Mask: net.CIDRMask(128, 128),
-		}
-		addr = v6Addr
-	}
 	// AddNetwork guarantees that Gateway string is a valid IPNet
 	gw := net.ParseIP(r.PodENISubnetGW)
 
-	var hostVethName string
+	var vethMetadata []driver.VirtualInterfaceMetadata
+	var podIPs []*current.IPConfig
+	var podInterfaces []*current.Interface
 	var dummyInterface *current.Interface
 
+	for index, ip := range r.IPAddress {
+
+		var hostVethName string
+
+		addr := parseIPAddress(ip)
```

Comment:
nit: this ip structure seems more like an "ipAllocation" instead of "ipAddress". i'd prefer to rename it.
also, rename this function to be something like getIPAddressFromIPAllocation to be specific